pkg/api/api.go

```diff
@@ -45,6 +47,13 @@ func NewDefinition() *Definition {
 
 // Validate validates proxy data
 func (d *Definition) Validate() (bool, error) {
+	r, _ := regexp.Compile("/^[A-Za-z0-9]+(?:-[A-Za-z0-9]+)*$/")
```

Comment:
good catch, but we can validate this with `govalidator` using `matches(pattern)` validator